### PR TITLE
Website: Fetch recipient details on partners page and transparency/finance page from firestore

### PR DESCRIPTION
**Pull request by DarkMenacer**
## Summary by CodeRabbit

- **Localization Updates**
	- Updated text strings for finances in German, English, French, and Italian locales.
	- Refined messaging about recipient support and program engagement.
	- Introduced "Active Recipients" metric across multiple languages.

- **Partner Management**
	- Simplified NGO identifiers.
	- Removed detailed recipient statistics from NGO entries.
	- Standardized partner data representation across languages.

- **Data Calculation**
	- Added new utility for calculating recipient statistics.
	- Enhanced transparency reporting with more precise recipient tracking.

- **User Interface**
	- Updated financial dashboard to display active recipient information.
	- Improved data rounding and presentation logic.
	- Modified `FundraiserBadge` to simplify user interaction by removing hover functionality.

### CODE REVIEW

File: shared/locales/de/website-partners.json [hunk 1]
```diff
@@ -283,6 +268,7 @@
 		"org-photo": ""
 	},
 	"sierra-leone-association-of-ebola-survivors": {
+		"firestore-id": "slaes",
```
Comment (mkue) [inline shared/locales/de/website-partners.json:271]: A key that is the same in every translation file, seems wrong. This file should be for strings that differ [across](url) languages.

File: shared/src/utils/stats/RecipientStatsCalculator.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+export class RecipientStatsCalculator {
+	constructor(readonly recipients: _.Collection<RecipientStatsEntry>) {}
+
+	/**
+	 * Calls the firestore database to retrieve the payments and constructs the
+	 * RecipientStatsCalculator with the flattened intermediate data structure.
+	 */
+	static async build(firestoreAdmin: FirestoreAdmin): Promise<RecipientStatsCalculator> {
+		const completeRecipientsData = await firestoreAdmin.collection<Recipient>(RECIPIENT_FIRESTORE_PATH).get();
+		const recipientStatsEntries = await Promise.all(
```
Comment (Gavriil-Tzortzakis) [inline shared/src/utils/stats/RecipientStatsCalculator.ts:35]: please filter out the test_recipients. This way we have less complex filtering.

File: shared/locales/en/website-partners.json [hunk 3]
```diff
@@ -32,15 +32,8 @@
...
-	"ngos": [
-		"aurora-foundation",
-		"jamil-&-nyanga-jaward-foundation",
-		"reachout-salone",
-		"equal-rights-alliance",
-		"united-polio-brothers-and-sisters",
-		"sierra-leone-association-of-ebola-survivors"
-	],
-	"aurora-foundation": {
+	"ngos": ["aurora", "jamil", "reachout", "equal-rights", "united-polio", "slaes"],
```
Comment (Gavriil-Tzortzakis) [inline shared/locales/en/website-partners.json:35]: Please move this to a dedicated .ts file (shared module)  as an enum.

File: shared/src/utils/stats/RecipientStatsCalculator.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		);
+		return {
+			total: totalRecipients.size(),
+			active: activeRecipients.size(),
+			former: formerRecipients.size(),
+			suspended: suspendedRecipients.size(),
+		};
+	};
+
+	allStats = (orgName?: string): RecipientStats => ({
```
Comment (Gavriil-Tzortzakis) [inline shared/src/utils/stats/RecipientStatsCalculator.ts:90]: Please make all the calculations of the organisation on this class. Then you can return a dictionary with the results already calculated to the calling component.

File: shared/src/utils/stats/RecipientStatsCalculator.test.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
+import { DocumentReference, Timestamp } from 'firebase-admin/firestore';
```
Comment (Gavriil-Tzortzakis) [inline shared/src/utils/stats/RecipientStatsCalculator.test.ts:1]: Please enrich this text with extra recipients and some edge cases.

**Review by Gavriil-Tzortzakis — CHANGES_REQUESTED** [review]

File: shared/src/utils/stats/RecipientStatsCalculator.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+						progr_status: recipientData.data().progr_status,
+						organisation: organisationSnapshot?.id,
+					};
+				}),
+		);
+		return new RecipientStatsCalculator(_(recipientStatsEntries));
+	}
+
+	totalRecipients = (): TotalRecipientsByStatus => {
+		const recipientsGroupedByProgStatus = _.groupBy(this.recipients.toJSON(), (x) => x.progr_status);
```
Comment (Gavriil-Tzortzakis) [inline shared/src/utils/stats/RecipientStatsCalculator.ts:53]: is this `toJSON` necessary?

**Review by Gavriil-Tzortzakis — APPROVED** [review]

**Review by ssandino — APPROVED** [review]
